pydoctor/sphinx_ext/__init__.py

```diff
@@ -1,3 +1,51 @@
 """
 Public and private extensions for Sphinx.
 """
+import os
+import subprocess
+
+
+def get_source_reference() -> str:
+    """
+    Return the reference for current build to be used when linking
+    the source files.
```

Comment:
And what do we do when we can't get a source reference?

Since this function is only called when `{source_reference}` was specified by the user, I think the correct response would be to fail the build if we can't get a reference. But it would be good to do that with a short and clear message rather than a stack trace. So I think this function should raise an exception and the caller should catch and report it.

Reply:
For example, when git is not found Sphinx build fails with this error.

```
Handler <function on_config_inited at 0x7fe5f7f13280> for event 'config-inited' threw an exception (exception: Failed to get git reference. [Errno 2] No such file or directory: 'git')

```

Reply:
Hmm, that's nowhere near as bad as a backtrace, but still slightly ugly.

In any case, I see you're moving the code from the extension into pydoctor itself, so the error reporting will have to be implemented in pydoctor as well, as we can't rely on Sphinx doing it anymore.